src/coreComponents/physicsSolvers/fluidFlow/StatOutputController.cpp

```diff
@@ -0,0 +1,153 @@
+  m_statistics( nullptr )
+{}
+
+void StatOutputController::initializePreSubGroups()
+{
+  Group & problemManager = this->getGroupByPath( "/Problem" );
+  DomainPartition & domain = problemManager.getGroup< DomainPartition >( "domain" );
+  Group & meshBodies = domain.getMeshBodies();
+
+  TasksManager & taskManager = this->getGroupByPath< TasksManager >( "/Tasks" );
+  OutputManager & outputManager = this->getGroupByPath< OutputManager >( "/Outputs" );
+  
+  std::vector< string > const groupNames = this->getSubGroupsNames();
+  m_statistics = &this->getGroup< TaskBase >( groupNames[0] );
+
+  forSubStats( [&]( auto & statistics ) {
+    using STATSTYPE = typename TYPEOFREF( statistics );
+    statistics.getSolver()->forDiscretizationOnMeshTargets( meshBodies, [&] ( string const &,
+                                                                              MeshLevel & mesh,
+                                                                              arrayView1d< string const > const & regionNames )
+    {
+      ElementRegionManager & elemManager = mesh.getElemManager();
+      for( string const & regionName : regionNames )
+      {
+        ElementRegionBase & region = elemManager.getRegion( regionName );
+        string const regionStatPath = GEOS_FMT( "{}/regionStatistics", region.getPath() );
+        typename STATSTYPE::RegionStatistics & regionStats = this->getGroupByPath< typename STATSTYPE::RegionStatistics >( regionStatPath );
+        string_array sourceTasks;
+
+        regionStats.forWrappers( [&]( WrapperBase const & wrapper )
+        { //PackCollection generation
+          string const taskManagerKey = GEOS_FMT( "packCollection{}{}", regionName, wrapper.getName());
+          PackCollection * packCollection = generatePackCollection( taskManager, taskManagerKey, regionStatPath, wrapper.getName());
+          m_packCollections.push_back( packCollection );
+          sourceTasks.emplace_back( GEOS_FMT( "{}/", packCollection->getPath()) );
+        } );
+
+        { //TimeHistory generation
+          string const outputManagerKey = GEOS_FMT( "compFlowHistory{}", regionName );
+          string const filename = GEOS_FMT( "generatedHDFStat{}", regionName );
+          TimeHistoryOutput * timeHistory = generateTimeHistory( outputManager, outputManagerKey, sourceTasks, filename );
+          m_timeHistories.push_back( timeHistory );
+        }
+      }
+    } );
+  } );
+
+}
+
+Group * StatOutputController::createChild( string const & childKey, string const & childName )
+{
+  GEOS_LOG_RANK_0( "Adding Statistics: " << childKey << ", " << childName );
+  std::unique_ptr< TaskBase > task = TaskBase::CatalogInterface::factory( childKey, childName, this );
+  return &this->registerGroup< TaskBase >( childName, std::move( task ) );
+}
+
+template< typename LAMBDA >
+void StatOutputController::forSubStats( LAMBDA lambda )
+{
+  forSubGroups< SinglePhaseStatistics,
+                CompositionalMultiphaseStatistics >( lambda );
```

Comment:
What error happens if (if there is any):
- no sub-stat `Group` exist,
- uncompatible `Group` used (any `Group` like `Box`),
- uncompatible `TaskBase` used (like `SolidMechanichStatistics`),

Reply:
@arng40

Reply:
1 & 2
```
***** ERROR
***** LOCATION: /data/pau901/VDG_GSI/ARNAUDD/geosRepos/codes/GEOS/src/coreComponents/physicsSolvers/fluidFlow/StatOutputController.cpp:133
***** Controlling expression (should be false): allowedChildTypes.count( childKey )==0
***** Rank 0:statController (staircase_3d.xml, l.168) must have one of the following components / CompositionalMultiphaseStatistics,SinglePhaseStatistics
```
3 
***** ERROR
***** LOCATION: /data/pau901/VDG_GSI/ARNAUDD/geosRepos/codes/GEOS/src/coreComponents/physicsSolvers/fluidFlow/StatOutputController.cpp:134
***** Controlling expression (should be false): allowedChildTypes.count( childKey )==0
***** Rank 0: Wrong children "X" in /Tasks/statController, allowed child types are CompositionalMultiphaseStatistics,SinglePhaseStatistics